gramineproject/gsc
# Update Intel's SGX PGP RSA key signing Intel repos

## Description of the changes 

The previous key expired on 2023-05-24 (which is today).

The new key was copied from https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key.

## How to test this PR? 

- Without this PR:
```
$ ./gsc build --insecure-args python test/generic.manifest

...
Step 4/30 : COPY intel-sgx-deb.key /

 ---> d1aa65346832
Step 5/30 : RUN echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main'     > /etc/apt/sources.list.d/intel-sgx.list     && apt-key add intel-sgx-deb.key

 ---> Running in 00693f0d7ef6
Warning: apt-key output should not be parsed (stdout is not a terminal)

OK
 ---> 2af1e0d2196d
Step 6/30 : RUN env DEBIAN_FRONTEND=noninteractive apt-get update     && env DEBIAN_FRONTEND=noninteractive apt-get install -y libsgx-dcap-quote-verify-dev

 ---> Running in 6b14bcfeccb8
Get:1 https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic InRelease [1240 B]
Err:1 https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY E5C7F0FA1C6C6C3C
Hit:2 http://security.ubuntu.com/ubuntu bionic-security InRelease
Hit:3 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:4 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:5 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Reading package lists...

W: GPG error: https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic InRelease: The following signatures couldn't be verifie
d because the public key is not available: NO_PUBKEY E5C7F0FA1C6C6C3C
E: The repository 'https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic InRelease' is not signed.
```

- With this PR, GSC works as usual.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gsc/152)
